website/static/js/apiApplication.js

```diff
@@ -0,0 +1,434 @@
+            home_url: this.homeUrl(),
+            callback_url: this.callbackUrl(),
+            client_id: this.clientId,
+            client_secret: this.clientSecret(),
+            owner: this.owner
+        };
+    }
+});
+
+/*
+ * Fetch data about applications
+ */
+var ApplicationDataClient = oop.defclass({
+    /*
+     * Create the client for server operations on ApplicationData objects.
+     * @param {Object} apiListUrl: The api URL for application listing/creation
+     */
+    constructor: function (apiListUrl) {
+        this.apiListUrl = apiListUrl;
+    },
+    _fetchData: function (url) {
+        var ret = $.Deferred();
+        var request = $osf.ajaxJSON('GET', url, {isCors: true});
+
+        request.done(function (data) {
+            ret.resolve(this.unserialize(data));
+        }.bind(this));
+
+        request.fail(function (xhr, status, error) {
+            ret.reject(xhr, status, error);
+        }.bind(this));
+
+        return ret.promise();
+    },
+    fetchList: function () {
+        return this._fetchData(this.apiListUrl);
+    },
+    fetchOne: function (url) {
+        return this._fetchData(url);
+    },
+    _sendData: function (appData, url, method) {
+        var ret = $.Deferred();
+
+        var payload = appData.serialize();
+        var request = $osf.ajaxJSON(method, url, {isCors: true, data: payload});
+
+        request.done(function (data) { // The server response will contain the newly created/updated record
+            ret.resolve(this.unserialize(data));
+        }.bind(this));
+        request.fail(function (xhr, status, error) {
+            ret.reject(xhr, status, error);
+        });
+        return ret.promise();
+    },
+    createOne: function (appData) {
+        var url = this.apiListUrl;
+        return this._sendData(appData, url, 'POST');
+    },
+    updateOne: function (appData) {
+        var url = appData.apiDetailUrl;
+        return this._sendData(appData, url, 'PATCH');
+    },
+    deleteOne: function (appData) {
+        var url = appData.apiDetailUrl;
+        return $osf.ajaxJSON('DELETE', url, {isCors: true});
+    },
+    unserialize: function (apiData) {
+        var result;
+        // Check return type: return one object (detail view) or list of objects (list view) as appropriate.
+        if (Array.isArray(apiData.data)) {
+            result = $.map(apiData.data, function (item) {
+                return new ApplicationData(item);
+            });
+        } else if (apiData.data) {
+            result = new ApplicationData(apiData.data);
+        } else {
+            result = null;
+        }
+        return result;
+    }
+});
+
+/*
+  ViewModel for List views
+ */
+var ApplicationsListViewModel = oop.defclass({
+    constructor: function (urls) {
+        this.apiListUrl = urls.apiListUrl;
+        // Set up data storage
+        this.appData = ko.observableArray();
+        this.sortByName = ko.computed(function () {
```

Comment:
Since this acts as a property rather than a method, it might make sense to name this `sortedByName`.